core/keyserver/validate.go

```diff
@@ -100,6 +103,34 @@ func validateListEntryHistoryRequest(in *pb.ListEntryHistoryRequest, currentRevi
 	return nil
 }
 
+// validateListUserRevisionsRequest checks the bounds on start and end revisions and returns an appropriate number of
+// revisions to return for this request.
+func validateListUserRevisionsRequest(in *pb.ListUserRevisionsRequest, offset, currentRevision int64) (int64, error) {
+	if in.StartRevision < 0 || in.StartRevision > currentRevision {
+		return 0, ErrInvalidStart
+	}
+	if in.EndRevision != 0 {
+		if in.EndRevision < in.StartRevision || in.EndRevision > currentRevision {
+			return 0, ErrInvalidEnd
+		}
+		currentRevision = in.EndRevision
+	}
+
+	revisions := int64(in.PageSize)
+	switch {
+	case in.PageSize < 0:
+		return 0, fmt.Errorf("Invalid page size")
+	case in.PageSize == 0:
+		revisions = int64(defaultPageSize)
```

Comment:
I would expect defaultPageSize to be maxPageSize. wdyt?

Reply:
Currently default page size is 16 entries (~2KB response data per request) and max is 2048 entries (~8MB response data per request). I think these are fine defaults to have for now, we can always change them or simplify later.